pkg/kubelet/pod/pod_manager.go

```diff
@@ -153,6 +158,9 @@ func (pm *basicManager) UpdatePod(pod *v1.Pod) {
 // lock.
 func (pm *basicManager) updatePodsInternal(pods ...*v1.Pod) {
 	for _, pod := range pods {
+		if pm.secretManager != nil {
```

Comment:
My mistake. Your assumption is right. There are some caveats which may cause extra fetches because pod manager doesn't distinguish update types:
 1. In the case where the only the status of a pod changes, kubelet gets a RECONCILE event in the sync loop. `podManager.UpdatePod` will be called.
 2. In the case where DeletionTimestamp is set (i.e., triggering a pod deletion),  kubelet gets a DELETE event in the sync loop. `podManager.UpdatePod` will be called.

(2) is probably ok since it happen only once per pod on deletion. (1) may cause significant more fetches during batch creation of pods. It may make sense to distinguish these update types in the pod manager so that you can avoid registering again.